### PR TITLE
tests/nested/core20/kernel-failover: add test for failed refresh of uc20 kernel

### PR DESCRIPTION
**Pull request by anonymouse64**
This exercises both the snapd logic for falling back properly and cleaning up
after a failed refresh, as well as pc-gadget grub.conf logic that rollbacks are
handled properly.

Eventually with resealing there should be additional things we check for here,
but for now it's sufficient to show that the kernel got rolled back successfully
and that we don't leave any intermediate state behind about the attempted
refresh.

Also some miscellaneous nested.sh improvements.

### CODE REVIEW

**Comment by sergiocazzolato** [comment]
@anonymouse64 thanks for this. The tests seems to be ok, but I have a question about where it should go.
Is there any reason to have it as part of the nested suite instead of the core suite? We do have the failover tests for core and snapd in core test suite.
Thanks

**Comment by anonymouse64** [comment]
@sergiocazzolato specifically, when we support resealing this test will have additional things to check for to demonstrate that we roll back and resealed against the correct thing after a failed test, if the core suite ever supports secure boot and TPM when booting, then this test could be there but until then I'd rather keep this test (or at least some variant of it) in the nested suites so that we can run it with full disk encryption enabled.

File: tests/nested/core20/kernel-failover/task.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  mv pc-kernel_*.snap panicking-initramfs-kernel.snap
+
+execute: |
+  # shellcheck source=tests/lib/nested.sh
+  . "$TESTSLIB/nested.sh"
+
+  # Copy the broken initramfs kernel snap to the UC20 VM
+  copy_remote panicking-initramfs-kernel.snap
+
+  # wait for snapd to be "available"
```
Comment (sergiocazzolato) [inline tests/nested/core20/kernel-failover/task.yaml:29]: I think we will need a set new nested-utils tool set with thinks like this

Comment (anonymouse64) [reply tests/nested/core20/kernel-failover/task.yaml:29]: sure makes sense

**Review by mvo5 — APPROVED** [review]
Nice one, thank you!

**Comment by mvo5** [comment]
Hm, looking at the logs it appears the new failover test is failing :/

**Comment by pedronis** [comment]
this test works as advertised if we get a user and to ssh at all

**Pull request merged**